src/transaction.ts

```diff
@@ -149,6 +163,10 @@ export class Transaction {
     return this.receiver;
   }
 
+  getGuardian(): IAddress | undefined {
```

Comment:
Can this be undefined though? Given this line in the constructor: `this.guardian = guardian || Address.empty();`

Reply:
Also, seeing the logic in the serializer that throws when transaction options say `it's a guarded tx`, but the `guardianSignature === 0x000[..]00`, we should either:
1. Throw here in the constructor when not providing a guardian (we don't have a setter for it, so if it's undefined, the transaction is invalid forever)
2. Add a setter and let the serializer go with `Address.empty()`
3. Add a setter and a public methods that checks that the tx has all the guardian fields set up (`return this.getOptions().isGuardedTx() && !this.guardian.equals(Address.empty())`). Then use this method in the serializer check